.github/workflows/build-and-tag-maven.yml

```diff
@@ -0,0 +1,100 @@
+
+      - name: Update pom files to the new version
+        id: update-pom-to-next-version
+        if: ${{ github.event_name == 'push' }}
+        uses: Alfresco/alfresco-build-tools/.github/actions/update-pom-to-next-pre-release@aae-8964-produce-alphas
+
+      - name: Set version env variable
+        if: ${{ github.event_name == 'push' }}
+        env:
+          VERSION: ${{ steps.update-pom-to-next-version.outputs.next-prerelease }}
+        run: |
+          echo "VERSION=$VERSION" >> $GITHUB_ENV
+
+      - name: Login to DockerHub Registry
+        uses: docker/login-action@v1
+        with:
+          registry: docker.io
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+
+      - name: Login to Quay.io Docker Registry
+        uses: docker/login-action@v1
+        with:
+          registry: quay.io
+          username: ${{ secrets.QUAY_USERNAME }}
+          password: ${{ secrets.QUAY_PASSWORD }}
+
+      - name: Define Maven Command
+        id: define_maven_command
+        run: |
+          if [ $DO_PUSH == 'true' ]
+          then
+            echo "::set-output name=command::deploy"
+          else
+            echo "::set-output name=command::verify"
+          fi
+        env:
+          DO_PUSH: ${{ github.event_name == 'push' }}
+      - name: Build and Test with Maven (and maybe Deploy)
+        run: mvn ${{ steps.define_maven_command.outputs.command }} ${{ env.MAVEN_CLI_OPTS}}
+        env:
+          MAVEN_CLI_OPTS: --show-version --no-transfer-progress --settings settings.xml -Dlogging.root.level=off -Dspring.main.banner-mode=off -Ddocker.skip
+          MAVEN_USERNAME: ${{ secrets.NEXUS_USERNAME }}
+          MAVEN_PASSWORD: ${{ secrets.NEXUS_PASSWORD }}
+      - name: Define Docker Push
+        id: define_docker_push
+        run: |
+          if [ $DO_PUSH == 'true' ]
+          then
+            echo "::set-output name=command::--push"
+          fi
+        env:
+          DO_PUSH: ${{ github.event_name == 'push' }}
+      - name: Docker Build (and maybe Push)
+        run: sh ./build-and-push-docker-images.sh
+        env:
+          PUSH_OPTION: ${{ steps.define_docker_push.outputs.command }}
+          TAG: ${{env.VERSION}}
+      - uses: Alfresco/alfresco-build-tools/.github/actions/git-commit-changes@v1.5.0
+        if: ${{ github.event_name == 'push' }}
+        with:
+          username: ${{ secrets.BOT_GITHUB_USERNAME }}
+          add-options: -u
+          commit-message: "release $VERSION"
+
+      - name: Create and push tag
+        if: ${{ github.event_name == 'push' }}
+        run: |
+          git tag -a $VERSION -m "Release version $VERSION"
+          git push origin $VERSION
+
+      - name: Slack Notification
+        if: ${{ failure() && github.event_name == 'push' }}
+        uses: rtCamp/action-slack-notify@v2
+        env:
+          SLACK_COLOR: ${{ job.status }}
+          SLACK_WEBHOOK: ${{ secrets.APA_SLACK_WEBHOOK }}
```

Comment:
This will definitely need to be extracted as an input parameter. Though I'm not sure the slack notif should be part of the reusable workflow.

Reply:
this conversation was somehow duplicated (?) --> resolving